### PR TITLE
Mouth/Eyes enums

### PR DESCRIPTION
**Pull request by kleril**
Added structs to actors that have logic for managing mouth textures so we're not just referring to textures by their raw index value.

### CODE REVIEW

**Review by cadmic — APPROVED** [review]

File: src/overlays/actors/ovl_En_Ma2/z_en_ma2.c [hunk 1]
```diff
@@ -51,6 +51,12 @@ static ColliderCylinderInit sCylinderInit = {
 
 static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
 
+typedef enum {
+    /* 0 */ MALON_ADULT_MOUTH_NEUTRAL,
+    /* 1 */ MALON_ADULT_MOUTH_SAD,
+    /* 2 */ MALON_ADULT_MOUTH_HAPPY
+} EnMa2MouthState;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Ma2/z_en_ma2.c:58]: This note applies to all actors modified here.

First: I personally dont find it necessary to call the enum as "mouth state". I prefer simply just calling them Mouth or Eyes, like you did in the enum prefix name.

Similarly, I dont think its necessary for the member variable to be called `mouthIndex`. I think its perfectly understandable as just `mouth` or `eyes`.
I understand this trend of calling it "xIndex" is somewhat common among lots of actors already. But if we agree this is nicer, we can make the changes where appropriate.

Second: You can prefix the enum name with the "friendly" actor name. Similar to how you use `MALON_ADULT_` in the enum prefix, you can similarly name the enum `MalonAdultMouth`, for example.

To summarize, I think this pattern overall is an improvement:
```
typedef enum {
    /* 0 */ MALON_ADULT_MOUTH_NEUTRAL,
    /* 1 */ MALON_ADULT_MOUTH_SAD,
    /* 2 */ MALON_ADULT_MOUTH_HAPPY
} MalonAdultMouth;

this->mouth = MALON_ADULT_MOUTH_HAPPY;
```
I did this recently with Player, even tho his eyes and mouth system is a bit more complicated than the rest of the actors

Comment (fig02) [reply src/overlays/actors/ovl_En_Ma2/z_en_ma2.c:58]: (AdultMalon probably reads nicer tbh. It looks like the limb enum name already uses this, so we should keep that consistent)

Comment (hiisuya) [reply src/overlays/actors/ovl_En_Ma2/z_en_ma2.c:58]: Peaked my interest seeing as I did some of Child Malon (`ma1`). But some of these changes would conflict with how some of the stuff within her code is doc'd. Would we want to make these changes (ie. `this->mouthIndex` -> `this->mouth` /  `EnMa2Mouth` -> `AdultMalonMouth`) and then reactively change `ma1` to match that as well right away (in this pr)? I supposed `ma3` as well by proxy. Or let the changes in and then change the other ones when it's gotten around to?

I'm not sure how this has been handled in other actors like heishi where so much is the same. Keeping them the same and up to date at the same time feels like the right thing to do but I'd understand if this was just something done, and the rest slowly falls into place. Not like it's pressing or anything, I'm mostly genuinely curious as to how the approach for this is handled. The PR does look good to me aside from my question here and your suggestion.

Comment (kleril) [reply src/overlays/actors/ovl_En_Ma2/z_en_ma2.c:58]: @fig02 agreed with the enum changes, will do a pass of those. As far as the member variable goes I'd argue "MouthTextureIndex" or some variant thereof would be best - there's a few actors with weird logic where just using "mouth" or "eye" would introduce ambiguity - see the [Bombchu Bowling Lady](https://github.com/zeldaret/oot/blob/main/src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c) for example.

@hiisuya I've been broadly going by "what's the style that the current file I'm looking at is abiding by" - been going for a jump from "completely undocumented" to "parse-able", unifying the naming I'm gonna leave for after Much Heated Debate which will shake out the 'correct' format and style to use.

Comment (fig02) [reply src/overlays/actors/ovl_En_Ma2/z_en_ma2.c:58]: That bombhu bowling girl one is interesting. I think its still fine personally if the variable that specifies the exact texture itself is `eyes`, that is the exact set of eyes you get shown. While the higher level variable can remain `eyeMode`, or possibly `eyeState`. Its describing why the actor has that set of eyes, not what eyes they have, right.

**Comment by fig02** [comment]
Closing due to inactivity. 
I recognize this PR got delayed due in part to us being undecided of how to handle these types of enums on a larger scale. Apologies for that.

Feel free to reopen this PR if you get back around to oot things and want to maintain the PR again
